### DISCUSSION TITLE
Direct access to the `root` parameter in the custom loader function for next/image

### DISCUSSION THREAD

**Discussion by addy-pathania** [vercel/next.js#33997] [Ideas]
### Describe the feature you'd like to request

All the built-in loader functions have access to the `root` param, when trying to access the same in the custom loader function, it doesn't work as expected.
All the built in loader have this passed: `return load({ root: configPath, ...loaderProps })`
[Refer to file](https://github.com/vercel/next.js/blob/c879fce0d465cd96d0d05e329c0a8a83f46e29d9/packages/next/client/image.tsx#L240)

Can we not directly have access to the `root` in the `ImageLoaderProps` and not extend it with `DefaultImageLoaderProps`
[Refer to file](https://github.com/vercel/next.js/blob/c879fce0d465cd96d0d05e329c0a8a83f46e29d9/packages/next/client/image.tsx#L28-L34)

### Describe the solution you'd like

```
export type ImageLoaderProps = {
  src: string
  width: number
  quality?: number
  root?: string
}
```

### Describe alternatives you've considered

N/A

**Comment by ambrauer** [comment]
Seems like this will be solved in the next release with the changes for #31065. [`ImageLoaderProps`](https://github.com/vercel/next.js/pull/33559/files#diff-ed7e3de801c4bc992b2796b6dd476f27320efef5d2911101e2ccb7b27c35866aR31) now has read-only access to the entire image config.

**Reply by addy-pathania** [reply]
Yes, seems like it did resolve it. We can close this now.
